alacritty/src/event.rs

```diff
@@ -1127,6 +1142,44 @@ impl input::Processor<EventProxy, ActionContext<'_, Notifier, EventProxy>> {
                     WindowEvent::KeyboardInput { input, is_synthetic: false, .. } => {
                         self.key_input(input);
                     },
+                    WindowEvent::Ime(ime) => match ime {
+                        Ime::Enabled => {
+                            self.ctx.display.ime_input.set_enabled(true);
+                        },
+                        // The text is commited thus should be written to PTY.
+                        Ime::Commit(text) => {
+                            self.ctx.display.ime_input.clear_preedit();
+                            if self.ctx.display.collect_damage() {
+                                // Damage the entire line with preedit.
+                                let line = self.ctx.display.ime_input.line();
+                                let point = Point::new(line, Column(0));
+                                let num_cols = self.ctx.terminal.columns() as u32 - 1;
+                                self.ctx.display.damage_from_point(point, num_cols);
+                            }
+
+                            // Don't suppress chars for commit.
+                            let mut suppress = false;
+                            mem::swap(self.ctx.suppress_chars, &mut suppress);
+                            text.chars().for_each(|ch| self.received_char(ch));
+                            *self.ctx.suppress_chars = suppress;
+                        },
+                        // This text is being composed, thus should be shown as inline input to
+                        // the user.
+                        Ime::Preedit(text, cursor_range) => {
+                            // If've got empty preedit we should start processing ordinary keyboard
+                            // input.
+                            if text.is_empty() && cursor_range.is_none() {
+                                self.ctx.display.ime_input.clear_preedit()
+                            } else {
+                                self.ctx.display.ime_input.set_preedit(text, cursor_range);
+                            }
```

Comment:
What platform are you on?

Reply:
Oh, I forgot that. I'm on macOS Monterey with M1 Mac.

Reply:
I've found one more bug. After I insert one or more korean character, ascii characters (like '0' or '+', etc.) are inserted two per key press.

I guess it is because when `ime_input` is enabled, ascii characters are processed both in `WindowEvent::Ime` and `WindowEvent::KeyboardInput`

Reply:
I've sent a fix upstream, I'll let you know once it merged and get resolved here.